sdk/test/metrics/sync_metric_storage_up_down_counter_test.cc

```diff
@@ -0,0 +1,327 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#include <memory>
+#ifndef ENABLE_METRICS_PREVIEW
+#  include "opentelemetry/common/key_value_iterable_view.h"
+#  include "opentelemetry/nostd/shared_ptr.h"
+#  include "opentelemetry/sdk/metrics/exemplar/histogram_exemplar_reservoir.h"
+#  include "opentelemetry/sdk/metrics/instruments.h"
+#  include "opentelemetry/sdk/metrics/state/sync_metric_storage.h"
+#  include "opentelemetry/sdk/metrics/view/attributes_processor.h"
+
+#  include <gtest/gtest.h>
+#  include <map>
+
+using namespace opentelemetry::sdk::metrics;
+using namespace opentelemetry::common;
+using M         = std::map<std::string, std::string>;
+namespace nostd = opentelemetry::nostd;
+
+class MockCollectorHandle : public CollectorHandle
+{
+public:
+  MockCollectorHandle(AggregationTemporality temp) : temporality(temp) {}
+
+  ~MockCollectorHandle() override = default;
+
+  AggregationTemporality GetAggregationTemporality(
+      InstrumentType /* instrument_type */) noexcept override
+  {
+    return temporality;
+  }
+
+private:
+  AggregationTemporality temporality;
+};
+
+class WritableMetricStorageTestFixture : public ::testing::TestWithParam<AggregationTemporality>
+{};
+
+TEST_P(WritableMetricStorageTestFixture, LongUpDownCounterSumAggregation)
+{
+  AggregationTemporality temporality      = GetParam();
+  auto sdk_start_ts                       = std::chrono::system_clock::now();
+  long expected_total_active_get_requests = 0;
+  long expected_total_active_put_requests = 0;
+  InstrumentDescriptor instr_desc = {"name", "desc", "1unit", InstrumentType::kUpDownCounter,
+                                     InstrumentValueType::kLong};
+  std::map<std::string, std::string> attributes_get = {{"RequestType", "GET"}};
+  std::map<std::string, std::string> attributes_put = {{"RequestType", "PUT"}};
+
+  std::unique_ptr<DefaultAttributesProcessor> default_attributes_processor{
+      new DefaultAttributesProcessor{}};
+  opentelemetry::sdk::metrics::SyncMetricStorage storage(
+      instr_desc, AggregationType::kSum, default_attributes_processor.get(),
+      ExemplarReservoir::GetNoExemplarReservoir(), nullptr);
+
+  storage.RecordLong(10l, KeyValueIterableView<std::map<std::string, std::string>>(attributes_get),
```

Comment:
define the number `10l` as a variable as it is shared between this and the following line?

Reply:
Thanks, done